tcms/bugs/tests/test_api.py

```diff
@@ -0,0 +1,51 @@
+# pylint: disable=attribute-defined-outside-init
+
+from xmlrpc.client import Fault as XmlRPCFault
+from xmlrpc.client import ProtocolError
+
+from tcms.bugs.models import Bug
+from tcms.bugs.tests.factory import BugFactory
+from tcms.management.models import Tag
+from tcms.rpc.tests.utils import APIPermissionsTestCase
+from tcms.tests import user_should_have_perm
+from tcms.tests.factories import TagFactory
+
+
+class TestAddTag(APIPermissionsTestCase):
+    """Test Bug.add_tag"""
+
+    permission_label = 'bugs.add_bug_tags'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.bug = BugFactory()
+        self.tag = TagFactory(name='rhel')
+
+    def verify_api_with_permission(self):
+        self.rpc_client.Bug.add_tag(self.bug.pk, self.tag.name)
+        self.assertIn(self.tag, Bug.objects.get(pk=self.bug.pk).tags.all())
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.Bug.add_tag(self.bug.pk, self.tag.name)
+
```

Comment:
The scenarios below are functional testing, not related to permissions. They should be in a separate test class.